src/hobbits-plugins/operators/Edit/edit.cpp

```diff
@@ -0,0 +1,236 @@
+    const Parameters &parameters,
+    QSharedPointer<PluginActionProgress> progress)
+{    
+    if (inputContainers.length() == 0) {
+        QList<QSharedPointer<BitContainer>> outputContainers;
+        qint64 outputSize = 10;
+        QSharedPointer<BitArray> outBits;
+        outBits = QSharedPointer<BitArray>(new BitArray(outputSize));
+
+        auto bitContainer = BitContainer::create(outBits);
+
+        outputContainers.append(bitContainer);
+        return OperatorResult::result(outputContainers, parameters);
+    }
+
+    if (inputContainers.length() > 1) {
+        return OperatorResult::error("Requires a single input bit container");
+        
+    }
+
+    QStringList invalidations = m_delegate->validate(parameters);
+    if (!invalidations.isEmpty()) {
+        return OperatorResult::error(QString("Invalid parameters passed to %1:\n%2").arg(name()).arg(invalidations.join("\n")));
+    }
+
+    int progressInt = 15;
+
+    progress->setProgressPercent(progressInt);
+
+    //get values from spin boxes
+    qint64 start = parameters.value("start").toInt();
+    qint64 length = parameters.value("length").toInt();
+
+
+    //get bits from text box
+    QString newBitsInRange = parameters.value("new_bits_in_range").toString();
+
+    //pte_bits is already filled with data from start to length
+    //
+    //auto bitContainer = inputContainers.first();
+    QList<QSharedPointer<BitContainer>> outputContainers;
+
+    
+
+    
+
+    //do something with bits here
+    //ex 0000 1111 -> 1111 0000
+
+    int end = start + length;
+    qint64 inputContainerSize = inputContainers.at(0)->size();
+    QSharedPointer<const BitArray> bits = inputContainers.at(0)->bits();
+
+    progressInt+=15; progress->setProgressPercent(progressInt);
+
+    int newBitsInRangeLength = newBitsInRange.length();
+
+    int unitSize = getUnitSize(parameters);
+
+    qint64 outputSize = inputContainerSize - (length * unitSize) + (newBitsInRangeLength * unitSize);
+    
+    QSharedPointer<BitArray> outBits = QSharedPointer<BitArray>(new BitArray(outputSize));
+
+    bits->copyBits(0, outBits.data(), 0, start * unitSize, BitArray::Copy);
+
+    bits->copyBits((qint64) end * unitSize, outBits.data(), (qint64)((qint64)end + newBitsInRangeLength - length)*unitSize, 
+        outputSize, BitArray::Copy);
+    progressInt+=45; progress->setProgressPercent(progressInt);
+
+    QSharedPointer<BitArray> newBits;
+
+    if (parameters.value("edit_type").toString() == "bit" ) { //bits
+        QRegExp re("^[0-1]*$");
+        if (!re.exactMatch(newBitsInRange))
+            return OperatorResult::error("You can only use 0 or 1 in bit mode.");
+       newBits  = parseBits( newBitsInRange);
+
+        
+    } else if (parameters.value("edit_type").toString() == "hex") { //hex
+        QRegExp re("^[a-f 0-9]*$");
+        if (!re.exactMatch(newBitsInRange))
+            return OperatorResult::error("Invalid hex character.");
```

Comment:
The `BitArray::fromString` method will produce better error string than this, so you should make the proposed `parseNewBits` method capable of returning any errors (maybe pass the `QStringList` through from here?)